src/test/java/frc/robot/subsystems/example/HoodTest.java

```diff
@@ -0,0 +1,36 @@
+package frc.robot.subsystems.example;
+
+import edu.wpi.first.units.Units;
+import frc.robot.subsystems.hood.Hood;
+import frc.robot.subsystems.hood.HoodIO;
+import frc.robot.subsystems.hood.HoodIOSim;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class HoodTest implements AutoCloseable {
+    private final Hood hood;
+    private final double Delta = 0.05;
+
+    public HoodTest() {
+        HoodIO hoodIO = new HoodIOSim();
+        Hood.initialize(hoodIO);
+        hood = Hood.getInstance();
+    }
+
+    @Test
+    public void testSetAngle() throws InterruptedException {
+        hood.setDefaultCommand(hood.setAngle(() -> Units.Radians.of(0.5).mutableCopy()));
+        Thread.sleep(3000);
+        Assertions.assertTrue(hood.atSetpoint());
+    }
+
+    @Test
+    public void testSetPower() throws InterruptedException {
+        hood.setDefaultCommand(hood.setPower(() -> 0.5));
+        Thread.sleep(3000);
+        Assertions.assertEquals(0.5, hood.getPower().in(Units.Volts), Delta);
```

Comment:
If you set a power of 0.5, the motor will run at 50% of the maximal voltage, which is 6.0 volts. I'm not sure it will run exactly at 6.0 volts however, so I don't know how valid this test is.

Reply:
Well actually, I know that it will work, which is why I think this test isn't especially effective.

Reply:
How is it not effective it tests if the setpower function reaches its setpoint

Reply:
Because behind the scenes, the applied voltage is the calculation for it. That means that if the motor uses position control, the applied voltage would be the PID calculation for it. And if it's simply voltage it would be that voltage. It's like that also for the real motor I believe.

Reply:
ok, so remove the test?

Reply:
Yeah I think it's best.

Reply:
It can also be modified to test the angle setting and not the power.

Reply:
eb1c199